src/plots/get_subplot_ids.js

```diff
@@ -0,0 +1,54 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+
+var Registry = require('../registry');
+
+/**
+ * Get the ids of the current subplots.
+ *
+ * @param {object} layout plotly full layout object.
+ * @param {string} type subplot type to look for.
+ *
+ * @return {array} list of ordered subplot ids (strings).
+ *
+ */
+module.exports = function getSubplotIds(layout, type) {
```

Comment:
I don't like this. Moving this one file broke a long circular chain in which `Plots` was required by `plots/cartesian/axis_ids.js`, resulting in a circular dependency.